package_bag/routines.py

```diff
@@ -0,0 +1,211 @@
+        """Validates the bag-info.txt file against the bagit profile"""
+        # TO DO: first vlaidation that "BagIt-Profile-Identifier" exists
+        new_bag = bagit.Bag(bag_path)
+        bag_info = new_bag.info
+        if "BagIt-Profile-Identifier" not in bag_info:
+            print("no bagit profile identifier")
+            # TO DO: return exception
+        else:
+            profile = bagit_profile.Profile(
+                new_bag.info.get("BagIt-Profile-Identifier"))
+            # TO DO: exception if cannot retrieve profile
+            return profile.validate_bag_info(new_bag)
+            # TO DO: exception if validation does not work
+
+    def get_data(self, bag):
+        """Saves bag data from the bag-info.txt file"""
+        new_bag = bagit.Bag(bag.bag_path)
+        bag.origin = new_bag.info.get('Origin')
+        bag.rights_id = new_bag.info.get('Rights-ID')
+        bag.end_date = new_bag.info.get('End-Date')
+        bag.save()
+        # TO DO: what is this returning?
+
+
+# how does something get sent from one bag to another? how does batching work?
+class GetRights(object):
+    """Send rights IDs to external service and receive JSON in return"""
+
+    def run(self):
+        url = "rights service url"
+        apikey = "rights service apikey"
+        has_rights = []
+        for bag in Bag.objects.filter(rights_data__isnull=True):
+            try:
+                rights_json = self.retrieve_rights(bag, url, apikey)
+                self.save_rights(bag, rights_json)
+                has_rights.append(bag)
+            except Exception as e:
+                print(e)
+        # get rights ids from database
+        # loop through rights ids
+        # retrieve rights
+        # save rights
+        return has_rights
+
+    def retrieve_rights(self, bag, url, apikey):
+        """Sends POST request to rights statement service, receives JSON in return"""
+        # url for get request
+        resp = post(
+            url,
+            data="data",  # TO DO: what data is sent to rights service? obviously includes rights ids
+            headers={
+                "Content-Type": "application/json",
+                "apikey": apikey,
+            },
+        )
+        # send get request
+        # get serialized rights back as json
+        # QUESTION: do we want to validate the json we get back?
+        # return saved json
+        return rights_json
+
+    def save_rights(self, bag, rights_json):
+        """Save JSON from rights statement service to database"""
+        bag.rights_data = rights_json
+        bag.save()
+        # save json...to file? rights.json?
+        pass
+
+
+class CreatePackage(object):
+    """Create JSON according to Ursa Major schema and package with bag"""
+
+    def run(self):
+        temp_dir = settings.TMP_DIR
+        dest_dir = settings.DEST_DIR
+        packaged = []
+        unpackaged = Bag.objects.filter(rights_data__isnull=False)
```

Comment:
This queryset will not get you the desired results...

Reply:
This still needs to be addressed - as written now, this queryset and the queryset for GetRights are exactly the same, which will result in some really weird behavior.